Sources/CacheAdvance/CacheHeaderHandle.swift

```diff
@@ -142,23 +144,27 @@ final class CacheHeaderHandle {
         return try handle.readDataUp(toLength: Int(FileHeader.expectedEndOfHeaderInFile))
     }
 
-    /// Checks if the expected header version matches the persisted header version.
+    /// Checks that the file's persisted metadata has the expected values.
     ///
     /// - Parameter persistedMetadata: The persisted header metadata.
-    /// - Returns: `true` if this object's header version matches that of `fileHeader`; otherwise `false`.
-    private func canOpenFile(with persistedMetadata: Metadata) -> Bool {
+    /// - Throws: `CacheAdvanceError.incompatibleHeader` if this object's header version does not match that of `fileHeader`.
+    private func checkFile(with persistedMetadata: Metadata) throws {
         // Our current file header version is 1.
```

Comment:
Is this still true? If we only have a single version then how could we be hitting this problem?

Reply:
Good q! Will answer in this thread: https://github.com/dfed/CacheAdvance/pull/50#discussion_r555927694